From: Jens Axboe <axboe@kernel.dk>
Date: Mon, 13 Jan 2025 10:27:54 -0700
Subject: nvme: fix bogus kzalloc() return check in nvme_init_effects_log()
Patch-mainline: v6.14-rc1
Git-commit: 170e086ad3997f816d1f551f178a03a626a130b7
References: git-fixes

nvme_init_effects_log() returns failure when kzalloc() is successful,
which is obviously wrong and causes failures to boot. Correct the
check.

Fixes: d4a95adeabc6 ("nvme: Add error path for xa_store in nvme_init_effects")
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Acked-by: Daniel Wagner <dwagner@suse.de>
---
 drivers/nvme/host/core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -3174,7 +3174,7 @@ static int nvme_init_effects_log(struct
 	struct nvme_effects_log *effects, *old;
 
 	effects = kzalloc(sizeof(*effects), GFP_KERNEL);
-	if (effects)
+	if (!effects)
 		return -ENOMEM;
 
 	old = xa_store(&ctrl->cels, csi, effects, GFP_KERNEL);
